functions.ipynb

```diff
@@ -130,6 +169,45 @@
     "You have the keyword `def`, the function's name, and its parameters in parentheses. The semi-colon tells Python that what comes next is the function's body."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "5579a33e-65d0-4edf-bbc1-14672f6c09a2",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "### Type hints\n",
+    "\n",
+    "Python doesn't require you to explicitly declare a variable's type. However, when defining functions, you can add **type hints** in the function's signature.\n",
+    "\n",
+    "Type hints improve code readability and can help catch potential type-related bugs early during development.\n",
+    "\n",
+    "::: {.callout-warning}\n",
+    "The Python interpreter **does not** enforce type hints and will not check them during runtime.\n",
+    "\n",
+    "Type hints are primarily intended for improving code readability and serving as documentation for developers. \n",
+    ":::\n",
+    "\n",
+    "---\n",
```

Comment:
That looks great to me. If you want to be pedantic you could say that type hints are also very useful for the IDE